numpy/__init__.pyi

```diff
@@ -1,15 +1,52 @@
-# very simple, just enough to start running tests
-#
 import builtins
-from typing import Any, Mapping, List, Optional, Tuple, Union
+
+from typing import (
+    Any, Dict, Iterable, List, Optional, Mapping, Sized,
+    SupportsInt, SupportsFloat, SupportsComplex, SupportsBytes, SupportsAbs,
+    Tuple, Union,
+)
+
+import sys
 
 from numpy.core._internal import _ctypes
 
 _Shape = Tuple[int, ...]
 
+# Anything that can be coerced into numpy.dtype. To avoid recursive
+# definitions, any nested fields are required to be castable to a dtype object
```

Comment:
Perhaps do `_DtypeLike_nested = Any  # TODO: wait for support for recursive types`, and then use it in place of `Any` below?